### PR TITLE
Hotfix for group indices in MGXS Pandas DataFrames

### PR DESCRIPTION
**Pull request by wbinventor**
This PR is a minor hotfix for the energy group indices in Pandas DataFrames for `openmc.mgxs.MGXS` objects. Previously, one would encounter an issue when using energy groups with energy group edges that differed by <= 10%. In these cases, one or more groups would erroneously be assigned the same energy group indices in the DataFrame produced by the `MGXS` class. This PR fixes that issue such that energy group indices are now appropriately assigned.

The reason for this is that the `Tally` class produces Pandas DataFrames with columns for "energy" or "energyout" filters populated by strings with the energy bounds. These strings truncated the energy bounds to a single decimal place. Since the `MGXS` class assigned energy group indices by inspecting these strings, some energy groups were assigned the same group index. Now, the code is simplified such that group indices are set directly in the DataFrame without a need to inspect the group bounds from strings from the `Tally` class DataFrame.

### CODE REVIEW

**Comment by paulromano** [comment]
Looks like the mgxs tests failed.

**Comment by wbinventor** [comment]
In my first commit for this PR I neglected to account for energyout filters. My latest push works for both energyin and energyout filters in `MGXS` objects. This affected some of the MGXS test results since the old "results_true.dat" included some extra whitespace in some of the Pandas DataFrame column names (of which I was unaware), which has now been removed. However, the numerical results remain identical.

File: tests/test_track_output/results_test.dat [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0"?>
+<VTKFile type="PPolyData" version="0.1" byte_order="LittleEndian" compressor="vtkZLibDataCompressor">
+  <PPolyData GhostLevel="0">
+    <PPoints>
+      <PDataArray type="Float32" Name="Points" NumberOfComponents="3"/>
+    </PPoints>
+    <Piece Source="poly_0.vtp"/>
+  </PPolyData>
+</VTKFile>
```
Comment (paulromano) [inline tests/test_track_output/results_test.dat]: You really like adding this file, huh?

Comment (wbinventor) [inline tests/test_track_output/results_test.dat]: Wow, yeah this has been a recurring issue :-) it's not that I like adding the file, it's just that the test suite doesn't clean it up. I removed it and added this file to .gitignore. Perhaps your or @smharper may want to use a better approach to clean this file up directly in the track output test.

**Pull request merged**